src/wasm-lib/kcl/src/lsp/kcl/mod.rs

```diff
@@ -285,6 +289,83 @@ impl Backend {
 
         completions
     }
+
+    pub async fn send_telemetry(&self) -> Result<()> {
+        // Get information about the user.
+        let user = self
+            .zoo_client
+            .users()
+            .get_self()
+            .await
+            .map_err(|e| anyhow::anyhow!(e.to_string()))?;
+
+        // Hash the user's id.
+        // Create a SHA-256 object
+        let mut hasher = sha2::Sha256::new();
+        // Write input message
+        hasher.update(user.id);
+        // Read hash digest and consume hasher
+        let result = hasher.finalize();
+        // Get the hash as a string.
+        let user_id_hash = format!("{:x}", result);
+
+        // Get the workspace folders.
+        // The key of the workspace folder is the project name.
+        let workspace_folders = self.workspace_folders();
+        let project_names: Vec<String> = workspace_folders.iter().map(|v| v.name.clone()).collect::<Vec<_>>();
+        // Get the first name.
+        let project_name = project_names
+            .first()
+            .ok_or_else(|| anyhow::anyhow!("no project names"))?
+            .to_string();
+
+        // Collect all the file data we know.
+        let mut buf = vec![];
+        let mut zip = zip::ZipWriter::new(std::io::Cursor::new(&mut buf));
+        for entry in self.current_code_map.iter() {
+            let file_name = entry.key().replace("file://", "").to_string();
+
+            let options = zip::write::FileOptions::default().compression_method(zip::CompressionMethod::Stored);
+            zip.start_file(file_name, options)?;
+            zip.write_all(entry.value())?;
+        }
+        // Apply the changes you've made.
+        // Dropping the `ZipWriter` will have the same effect, but may silently fail
+        zip.finish()?;
+        drop(zip);
+
+        // Send the telemetry data.
+        self.zoo_client
+            .meta()
+            .create_event(
+                vec![kittycad::types::multipart::Attachment {
+                    // Clean the URI part.
+                    name: "attachment".to_string(),
+                    filename: None,
```

Comment:
Could we make filename something like "attachment.zip"? When it gets uploaded with no filename it's saved as "blob" on google storage which makes it a bit more of a pain to use even though the data is fine.

Reply:
ah gotcha will do